### PR TITLE
Gok-425 | Add. read xml and plain text based QR data

### PR DESCRIPTION
**Pull request by SanoferSameera**
Co-authored-by:yenugukeerthana <keerthana.yenugu@thoughtworks.com>

### CODE REVIEW

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt
Comment (sweetyTW) [inline app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt:1]: can we add "parser" in the file name? like "AadhaarQROutputParser"?

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        scannedAadhaarCardInfo.gender = result[AadhaarDataAttributes.AADHAAR_GENDER_ATTR]
+        scannedAadhaarCardInfo.dateOfBirth = result[AadhaarDataAttributes.AADHAAR_DOB_ATTR]
+        scannedAadhaarCardInfo.villageTownCity = result[AadhaarDataAttributes.AADHAAR_VTC_ATTR]
+        scannedAadhaarCardInfo.subDistrict = result[AadhaarDataAttributes.AADHAAR_SUB_DIST_ATTR]
+        scannedAadhaarCardInfo.district = result[AadhaarDataAttributes.AADHAAR_DIST_ATTR]
+        scannedAadhaarCardInfo.state = result[AadhaarDataAttributes.AADHAAR_STATE_ATTR]
+        scannedAadhaarCardInfo.pinCode = result[AadhaarDataAttributes.AADHAAR_PC_ATTR]
+    }
+
+    fun getAadhaarCardInfo(): AadhaarCardInfo {
```
Comment (sweetyTW) [inline app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt]: Is this function "getAadhaarCardInfo" getting used anywhere? if not can we remove it

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt
Comment (SanoferSameera) [reply app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt:1]: @sweetyTW yeah, I'm thinking of separating the class as it does two functionality xml and plain text

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt (hunk 1 above)
Comment (SanoferSameera) [reply app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt]: not used now, but it is needed to read and display the output info

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import java.io.StringReader
+import javax.xml.parsers.DocumentBuilderFactory
+
+
+class AadhaarXmlAndTextQr(scanData: String?) {
+    private var scannedAadhaarCardInfo: AadhaarCardInfo = AadhaarCardInfo()
+
+    init {
+        if(isXmlData(scanData)) {
+            processXmlData(scanData)
```
Comment (sweetyTW) [inline app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt:18]: Can we handle a scenario where something random is scanned which is not aadhaar?

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt (hunk 1 above)
Comment (sweetyTW) [reply app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt]: if its gonna used in near future we can keep it.

File: app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt
Comment (sweetyTW) [reply app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt:1]: In this PR itself?

Comment (SanoferSameera) [reply app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt:1]: yes. I'm on it. will let u know

Comment (SanoferSameera) [reply app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt:1]: done. pls check now

**Review by sweetyTW — APPROVED** [review]

**Pull request merged**